addons/website_livechat/models/discuss_channel.py

```diff
@@ -21,7 +21,7 @@ def channel_pin(self, pinned=False):
          delete discuss_channel as not useful to keep empty chat
          """
         super().channel_pin(pinned=pinned)
-        if self.livechat_active and not self.message_ids:
+        if self.channel_type == "livechat" and not self.livechat_end_dt and not self.message_ids:
```

Comment:
Maybe I'm missing something but I wonder why do we bother to check livechat_end_dt. It feels if there is no message, we can unlink whether it is closed or not?

Reply:
I believe it's for chat request which doesn't have any messages initially but should still be pinned.

Reply:
And chat request won't have livechat_end_dt set either?